docs/development/development.md

```diff
@@ -0,0 +1,258 @@
+
+The code that handles this functionality can be found in `pkg/dependencies/download.go` at line 43, where the files are downloaded, and in `pkg/x/net/client.go` at line 65, where caching is managed. If you want to modify the caching behavior, you can intervene on these files to add custom logic, such as version validation or timestamp checks to determine when to update the cache.
+
+</details>
+
+---
+
+### **How do you use `furyctl` in airgapped environments, and what flags are useful in these cases?**
+
+<details>
+<summary>Answer</summary>
+
+In airgapped environments, where no external connection is available to download binaries and dependencies, everything must be pre-downloaded and available locally. Binaries and resources can be copied manually into the `.furyctl` folder via tools like Ansible or committed directly to the project.
+
+To use `furyctl` in these environments, the following flags should be used:
+
+- `--distro-location`: This flag specifies the local path of the downloaded distribution, allowing `furyctl` to use the local version instead of attempting to download it.
+- `--skip-deps-download`: This flag skips downloading additional dependencies or binaries from external sources, ensuring that everything is used from the cache or the local distribution.
+</details>
+
+---
+
+### **How does `furyctl` apply patches to distribution versions, and does it download new dependency versions or use the initial ones?**
+
+<details>
+<summary>Answer</summary>
+
+The patches are applied in a way that resembles a "copy-paste" over the downloaded distribution files. When a patch (e.g., for `kfd.yaml`) is provided, it is applied directly on top of the version of the distribution already downloaded and available on the system. Before applying the patch, no new dependency versions are downloaded; instead, the initial version (the one downloaded initially) is used, and the specified changes are overwritten on top.
+
+This approach is helpful for applying local customizations without needing to repeat the entire dependency download process.
+
+</details>
+
+---
+
+### **How does `furyctl` validate the schema of the `furyctl.yaml` file, what is done by our library, and what is handled by the official library?**
+
+<details>
+<summary>Answer</summary>
+
+The schema of the `furyctl.yaml` file is validated using the Go library `https://github.com/santhosh-tekuri/jsonschema`. This library allows for validating a JSON/YAML file against a defined schema, ensuring that the structure of the data is correct and conforms to the specifications.
+
+Our library does not directly intervene in this validation step but merely downloads and provides the correct schema via `fury-distribution`, which is then used for the validation process.
```

Comment:
And what about the change we did for the `date` type format? The conflict between "our" validate and the library validation

Reply:
The change was made on https://github.com/sighupio/go-jsonschema, so it should be documented either on fury-distribution or directly on go-jsonschema.